lib/net.js

```diff
@@ -1392,6 +1395,14 @@ Server.prototype.listen = function() {
   var options = normalized[0];
   var cb = normalized[1];
 
+  if (this._serverListenHasBeenCalled === true) {
+    var er = new Error('listen method has been called more than once.');
+    er.code = 'EINVAL';
+    return this.emit('error', er);
+  }
+
+  this._serverListenHasBeenCalled = true;
```

Comment:
Rather than introducing a new state variable for tracking, just look for the existence of the `_handle`. If it is not `undefined` or `null`, the server is listening, otherwise it is not.